src/buffer/out/TextColor.cpp

```diff
@@ -64,7 +64,7 @@ bool TextColor::CanBeBrightened() const noexcept
 
 bool TextColor::IsLegacy() const noexcept
 {
-    return IsIndex16() || (IsIndex256() && _index < 16);
+    return (IsIndex16() || IsIndex256()) && _index < 16;

```

Comment:
This allows for better compiler optimizations, because Index16 and Index256 are two neighboring values in the enum. The compiler optimize the || away entirely.

Reply:
Since this is a enum class, there are no implicit operators to turn them into integers. I don't really want to write the rather verbose code required to do `static_assert` checks, so I've instead opted to add a comment to the declaration instead, which points this out.